### PR TITLE
use drake cmakelists

### PR DESCRIPTION
**Pull request by manuelli**
- bump drake to master and use CMake build that wraps bazel
- remove drake-schunk-driver built with bazel, rely on ros package version
- build libbot from source since it's gone from drake

This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/spartan/233)

### CODE REVIEW

**Comment by manuelli** [comment]
@gizatt looks like the build is failing due to an issue with `VHACD` which is part of your `trimesh` external project. Do you know what the issue is? Can you either fix it or disable the `WITH_TRIMESH` flag for jenkins.

**Comment by gizatt** [comment]
I'm seeing an Apriltags build error, not VHACD, in the Jenkins build log:

```
21:29:49 CMakeFiles/drc-apriltags-driver.dir/apriltags_driver.cpp.o: In function `AprilTagDetector::AprilTagDetector(getopt*)':
21:29:49 apriltags_driver.cpp:(.text._ZN16AprilTagDetectorC2EP6getopt[_ZN16AprilTagDetectorC5EP6getopt]+0x2c2): undefined reference to `YAML::LoadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
21:29:49 apriltags_driver.cpp:(.text._ZN16AprilTagDetectorC2EP6getopt[_ZN16AprilTagDetectorC5EP6getopt]+0x317): undefined reference to `YAML::detail::node_data::begin()'
21:29:49 apriltags_driver.cpp:(.text._ZN16AprilTagDetectorC2EP6getopt[_ZN16AprilTagDetectorC5EP6getopt]+0x3ec): undefined reference to `YAML::detail::node_data::end()'
21:29:49 collect2: error: ld returned 1 exit status
21:29:49 src/CMakeFiles/drc-apriltags-driver.dir/build.make:170: recipe for target 'bin/drc-apriltags-driver' failed
21:29:49 make[8]: *** [bin/drc-apriltags-driver] Error 1
21:29:49 CMakeFiles/Makefile2:117: recipe for target 'src/CMakeFiles/drc-apriltags-driver.dir/all' failed
21:29:49 make[7]: *** [src/CMakeFiles/drc-apriltags-driver.dir/all] Error 2
21:29:49 Makefile:127: recipe for target 'all' failed
21:29:49 make[6]: *** [all] Error 2
21:29:49 CMakeFiles/apriltags_driver.dir/build.make:112: recipe for target 'src/apriltags_driver-stamp/apriltags_driver-build' failed
21:29:49 make[5]: *** [src/apriltags_driver-stamp/apriltags_driver-build] Error 2
21:29:49 CMakeFiles/Makefile2:404: recipe for target 'CMakeFiles/apriltags_driver.dir/all' failed
21:29:49 make[4]: *** [CMakeFiles/apriltags_driver.dir/all] Error 2
21:29:49 make[4]: *** Waiting for unfinished jobs....
```

The old use-Drake-CMakelists upgrade had a VHACD failure, so I would expect to see one here (it has to be related to an environment change due to the top-level CMake changes). It's OK to turn it off to get this old branch landed if you can verify the error still happens -- change the default in the top-level CMakelist and change the Jenkins CMake build command in `setup/build/run_jenkins.sh`

**Comment by manuelli** [comment]
The build is now failing with 

```

CMakeFiles/drc-apriltags-driver.dir/apriltags_driver.cpp.o: In function `AprilTagDetector::AprilTagDetector(getopt*)':
21:29:49 apriltags_driver.cpp:(.text._ZN16AprilTagDetectorC2EP6getopt[_ZN16AprilTagDetectorC5EP6getopt]+0x2c2): undefined reference to `YAML::LoadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
21:29:49 apriltags_driver.cpp:(.text._ZN16AprilTagDetectorC2EP6getopt[_ZN16AprilTagDetectorC5EP6getopt]+0x317): undefined reference to `YAML::detail::node_data::begin()'
21:29:49 apriltags_driver.cpp:(.text._ZN16AprilTagDetectorC2EP6getopt[_ZN16AprilTagDetectorC5EP6getopt]+0x3ec): undefined reference to `YAML::detail::node_data::end()'
21:29:49 collect2: error: ld returned 1 exit status
21:29:49 src/CMakeFiles/drc-apriltags-driver.dir/build.make:170: recipe for target 'bin/drc-apriltags-driver' failed
21:29:49 make[8]: *** [bin/drc-apriltags-driver] Error 1
21:29:49 CMakeFiles/Makefile2:117: recipe for target 'src/CMakeFiles/drc-apriltags-driver.dir/all' failed
21:29:49 make[7]: *** [src/CMakeFiles/drc-apriltags-driver.dir/all] Error 2
21:29:49 Makefile:127: recipe for target 'all' failed
21:29:49 make[6]: *** [all] Error 2
```

it looks like apriltags driver is not finding yaml-cpp. @jamiesnape I am guessing this related to [this](https://github.com/RobotLocomotion/drake/pull/7453) PR? This is in the director build and I guess drake no longer supplies yaml-cpp linker flags.

**Comment by manuelli** [comment]
@jamiesnape so I guess drake still has yaml-cpp,. I am not sure why the apriltags-driver is suddenly having problems linking against yaml-cpp. Do you think it is related to the bug in yaml-cpp mentioned in https://github.com/RobotLocomotion/drake/pull/7602? As a short term fix we can disable apriltags_drivers as no one is using them.

**Comment by jamiesnape** [comment]
Looks like the variable `yaml-cpp_LIBRARIES` referred to here is called `YAML_CPP_LIBRARIES` in the system installed library:  target_link_libraries(${APP_NAME} ${Eigen3_LIBRARIES} ${yaml-cpp_LIBRARIES} ${libbot_LIBRARIES})

**Comment by manuelli** [comment]
Ok thanks @jamiesnape that makes sense. I will implement a fix upstream in the apriltags driver or disable it. It seems like adding `yaml-cpp` to target_link_libraries fixes things.

**Comment by jamiesnape** [comment]
Yes:
```
# - Config file for the yaml-cpp package
# It defines the following variables
#  YAML_CPP_INCLUDE_DIR - include directory
#  YAML_CPP_LIBRARIES    - libraries to link against

...

# These are IMPORTED targets created by yaml-cpp-targets.cmake
set(YAML_CPP_LIBRARIES "yaml-cpp")
```

File: src/ReachabilityAnalyzer/CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,8 @@ include_directories(${Eigen3_INCLUDE_DIR})
 find_package(VTK REQUIRED)
 include_directories(${VTK_INCLUDE_DIRS})
 
+find_path(yaml-cpp REQUIRED)
```
Comment (jamiesnape) [inline src/ReachabilityAnalyzer/CMakeLists.txt]: `find_package` ?

Comment (manuelli) [reply src/ReachabilityAnalyzer/CMakeLists.txt]: Whoops, typo on my part. Just pushed a fix.

**Comment by manuelli** [comment]
I just tested the IK issue from #234 and it seems to be working fine. I think this is ready to be merged. @peteflorence @gizatt

**Comment by gizatt** [comment]
:lgtm:

---

Reviewed 1 of 3 files at r1, 4 of 5 files at r2, 1 of 1 files at r3, 1 of 2 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/ReachabilityAnalyzer/CMakeLists.txt, line 26 at r5](https://reviewable.io:443/reviews/robotlocomotion/spartan/233#-L3FdCPCJzp4lVTrvCY7:-L3FdCPCJzp4lVTrvCY8:b4docg) ([raw file](https://github.com/robotlocomotion/spartan/blob/05a5230d8956d1dca7c25f295ac2a12585283c27/src/ReachabilityAnalyzer/CMakeLists.txt#L26)):*
> ```CMake
>     ${VTK_LIBRARIES}
>     RemoteTreeViewerWrapper
>     yaml-cpp)
> ```

I think more "correct" would be ${YAML_CPP_LIBRARIES}, but this is probably safer since that variable name hasn't been consistent across system-vs-local Yaml in the recent past...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/233#-:-L3FdeYTYkl0h8ZzNQBr:bnfp4nl)*

**Comment by manuelli** [comment]
@jamiesnape @fbudin69500 I was under the impression that the `bot_core` python package and `bot_core` lcmtypes were provided by libbot correct? From what I understand `drake` no longer has libbot, but does it still build those lcmtypes? We have somehow ended up with two versions of the `bot_core` python package installed. One is in


```
spartan/build/install/lib/python2.7/site-packages/bot_core
```

and the other in

```
spartan/build/install/lib/python2.7/dist-packages/bot_core
```

They have a very different set of lcmtypes. In particular the one in `site-packages` has many more, including `position_3d_t`. Is one of these being installed by `drake` and the other by the `libbot` CMake External project? This is all related to issue #236.

**Comment by jamiesnape** [comment]
> They have a very different set of lcmtypes. In particular the one in site-packages has many more, including position_3d_t. Is one of these being installed by drake and the other by the libbot CMake External project?

Yes, there was a fork of the `bot_core` lcmtypes to `bot_core_lcmtypes` https://github.com/openhumanoids/bot_core_lcmtypes.

**Comment by jamiesnape** [comment]
(you probably want the `site-packages` version first in your `PYTHONPATH`)

**Comment by manuelli** [comment]
Yeah I am coming to that conclusion. Somehow when I launch files a particular way the `dist-packages` gets put in front. Is there a way to avoid having two copies of `bot_core` installed?

**Comment by jamiesnape** [comment]
> Somehow when I launch files a particular way the dist-packages gets put in front.

Via `directorPython`?

**Comment by manuelli** [comment]
Looks like the `bot-procman-sheriff` executable is modifying the PYTHONPATH.

```
#!/bin/sh
export PYTHONPATH=/home/manuelli/spartan/build/install/lib/python2.7/dist-packages:/home/manuelli/spartan/build/install/lib/python2.7/site-packages:${PYTHONPATH}
exec /usr/bin/python -m bot_procman.sheriff_gtk.sheriff_gtk $*
```

I am guessing that `bot-procman-sheriff` is installed by `libbot`.

**Comment by manuelli** [comment]
`bot-procman-sheriff` is modifying the pythonpath. Looks like that logic is happening [here](https://github.com/RobotLocomotion/libbot2/blob/master/bot2-procman/python/bot-procman-sheriff.in)

**Comment by jamiesnape** [comment]
I guess we should be appending rather than prepending, I will PR a change.

**Comment by manuelli** [comment]
Thanks. Once we do that should we update `spartan` to point to `libbot/master`?

**Comment by jamiesnape** [comment]
Yes, https://github.com/RobotLocomotion/libbot2/commit/0813e5a66a09fcbdaf3c7fae69dc05eeb1223749.

**Pull request merged**